docs/reference-guides/filters/block-filters.md

```diff
@@ -81,6 +81,31 @@ wp.hooks.addFilter(
 
 The following filters are available to change the behavior of blocks while editing in the block editor.
 
+### `blocks.convertLegacyBlockNameAndAttributes`
+
+Filter the name and/or attributes of the block while the parsing process.
+Use this hook as an opportunity to change the name and/or attributes of the block.
+The function bound to the hook will run both at the parser level for previous content and to convert such blocks used in Custom Post Types templates.
+
+Take advantage of it, for instance, to convert legacy blocks to their canonical form:
+
+_Example:_
+
+```js
+wp.hooks.addFilter(
+	'blocks.convertLegacyBlockNameAndAttributes',
+	'my-plugin/book-update-block-name',
+	function ( [ blockType, blockAttributes ] ) {
```

Comment:
Should it be `blockName` instead of `blockType`?

Reply:
Have you considered simplifying the function param to take two params instead of a tuple?

```js
function ( name, attributes ) {
    return [ name, attributes ];
}
```

Reply:
yes, I did, and I agree with this change. But wanted to keep the initial implementation. Let me update it.

Reply:
I'm not sure whether there is a straightforward to change what the bound function expects and what returns without introducing significant changes :-/

Reply:
> Should it be blockName instead of blockType?

addressed here https://github.com/WordPress/gutenberg/pull/42712/commits/e2afddf9b26687989e472795a75c6670217a8e07

Reply:
You are right. It has to stay as is as the return value gets passed to the next filter 👍🏻 

I'm still not clear what will happen when several filters want to modify the same block. We can leave the decision to also pass the original block name or attributes as additional params for later.

Reply:
How/where should it store the original name?
Is it something we'd like, or even more we need, to delegate to the bound function to the filter? 

```js
addFilter(
	'blocks.migrateBlockType',
	'my-plugin/book-update-block-name',
	function ( [ name, attributes, prevName ] ) {
		// Convert 'my-plugin/book-block' block in existing content to 'my-plugin/book'.
		if ( 'my-plugin/book-block' !== name ) {
			// return the block original data
			return [ name, attributes, prevName ];
		}

		// return the new block with the previous name in the third parameter.
		return [ 'my-plugin/book', attributes, name ];
	}
);
```

Above, the prevName is provided by the filter, but to conserve the chain, the function will need to return the previous name too.

Reply:
Let's skip that for now.